python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -629,10 +629,17 @@ def _submit_runs_and_update_backfill_in_chunks(
         instance.update_backfill(updated_backfill)
 
     if not mid_iteration_cancel_requested:
-        check.invariant(
-            submitted_partitions == asset_backfill_iteration_result.backfill_data.requested_subset,
-            "Did not submit run requests for all expected partitions",
-        )
+        if submitted_partitions != asset_backfill_iteration_result.backfill_data.requested_subset:
+            missing_partitions = list(
+                (
+                    asset_backfill_iteration_result.backfill_data.requested_subset
+                    - submitted_partitions
+                ).iterate_asset_partitions()
+            )
+            check.failed(
+                "Did not submit run requests for all expected partitions. \n\nPartitions not"
+                f" submitted: {missing_partitions}",
```

Comment:
If there are thousands of partitions, this could be a pretty unwieldy error message.

Reply:
This check shouldn't be firing unless if there's an error in our asset backfill logic... I think it is unlikely for many partitions to be unsubmitted